packages/components/pie-toast/test/component/pie-toast.spec.ts

```diff
@@ -1,317 +1,256 @@
+                const footer = page.getByTestId(toast.selectors.footer.dataTestId);
+                const leadingAction = page.getByTestId(toast.selectors.leadingAction.dataTestId);
 
                 // Assert
-                expect(toast).toBeVisible();
-                expect(footer).toBeVisible();
-                expect(leadingAction).toBeVisible();
+                await expect(toastComponent).toBeVisible();
+                await expect(footer).toBeVisible();
+                await expect(leadingAction).toBeVisible();
             });
         });
 
         test.describe('leadingAction', () => {
-            test('should show the leadingAction when provided and if multiline is false', async ({ mount, page }) => {
+            test('should show the leadingAction when provided and if multiline is false', async ({ page }) => {
                 // Arrange
-                await mount(PieToast, {
-                    props: {
-                        isMultiline: false,
-                        leadingAction: mainAction,
-                    } as ToastProps,
-                });
-
-                // Wait for the component to render alongside its animations
-                await page.waitForTimeout(COMPONENT_RENDER_DURATION_MS);
+                const toastPage = new BasePage(page, 'toast');
+                const props: ToastProps = {
+                    isMultiline: false,
+                    leadingAction: mainAction,
+                };
+                await toastPage.load({ ...props });
 
                 // Act
-                const toast = page.locator(componentSelector);
-                const leadingAction = page.locator(leadingActionSelector);
+                const toastComponent = page.getByTestId(toast.selectors.container.dataTestId);
+                const leadingAction = page.getByTestId(toast.selectors.leadingAction.dataTestId);
 
                 // Assert
-                expect(toast).toBeVisible();
-                expect(leadingAction).toBeVisible();
+                await expect(toastComponent).toBeVisible();
+                await expect(leadingAction).toBeVisible();
             });
 
-            test('should not show the footer if leadingAction is not provided', async ({ mount, page }) => {
+            test('should not show the footer if leadingAction is not provided', async ({ page }) => {
                 // Arrange
-                await mount(PieToast, {
-                    props: {
-                        isMultiline: true,
-                    } as ToastProps,
-                });
-
-                // Wait for the component to render alongside its animations
-                await page.waitForTimeout(COMPONENT_RENDER_DURATION_MS);
+                const toastPage = new BasePage(page, 'toast');
+                const props: ToastProps = {
+                    isMultiline: true,
+                    leadingAction: undefined,
```

Comment:
I think this still counts as passing the prop even if it has an undefined value. I'd probably to modify the Storybook file so `leadingAction` has a default value of `''`, allowing you to remove it here from the props object.